spec/lib/shanty/plugin_dsl_spec.rb

```diff
@@ -40,4 +49,18 @@
       expect(subject.instance_variable_get(:@tags)).to contain_exactly(:plugin_name, :bar, :lux)
     end
   end
+
+  describe('.provides_config') do
+    it('can set a default value for a config key') do
+      subject.provides_config(:nic, 'cage')
+
+      expect(subject.config[:nic]).to eql('cage')
+    end
+
+    it('can set set an expectation for a config key') do
+      subject.provides_config(:nic)
+
+      expect(subject.config[:nic]).to be_nil
```

Comment:
Oh no - this would be in addition. Have two assertions in this test. The problem is that it could be `nil` but only because the config wasn't actually stored at all, or it could be `nil` because the behaviour you describe is taking place. I suspect you want to check the `nil` case and also that under the covers, the key is being stored.